projects/valtimo/components/src/lib/components/select/select.component.ts

```diff
@@ -0,0 +1,180 @@
+    this._listItems$.next(
+      value?.map(
+        item =>
+          ({content: item.text || item.translationKey, id: item.id, selected: false}) as ListItem
+      ) || []
+    );
+  }
+  @Input() public defaultSelection!: SelectItem;
+  @Input() public defaultSelectionId!: string;
+  @Input() public defaultSelectionIds!: Array<string>;
+  @Input() public disabled = false;
+  @Input() public multiple = false;
+  @Input() public margin = false;
+  @Input() public widthInPx!: number;
+  @Input() public notFoundText!: string;
+  @Input() public clearAllText!: string;
+  @Input() public name = '';
+  @Input() public title = '';
+  @Input() public titleTranslationKey = '';
+  @Input() public clearSelectionSubject$!: Observable<void>;
+  @Input() public tooltip = '';
+  @Input() public required = false;
+  @Input() public loading = false;
+  @Input() public loadingText = '';
+  @Input() public placeholder = '';
+  @Input() public smallMargin = false;
+  @Input() public carbonTheme = 'g10';
+
+  @Output() public selectedChange: EventEmitter<SelectedValue> = new EventEmitter();
+
+  public readonly selected$ = new BehaviorSubject<SelectedValue>('');
+  private readonly _listItems$ = new BehaviorSubject<Array<ListItem>>([]);
+
+  public readonly listItems$: Observable<Array<ListItem>> = combineLatest([
+    this._listItems$,
+    this.selected$,
+    this.translateService.stream('key'),
+  ]).pipe(
+    map(([listItems, selected]) =>
+      listItems.map(listItem => {
+        const translation = this.translateService.instant(listItem.content);
+        const isSelected: boolean = Array.isArray(selected)
+          ? selected.includes(listItem.id)
+          : selected === listItem.id;
+
+        return {
+          ...listItem,
+          selected: isSelected,
+          content: translation !== listItem.content ? translation : listItem.content,
+        };
+      })
+    ),
+    tap(items => console.log(items))
+  );
+
+  private _selectedSubscription!: Subscription;
+  private _clearSubjectSubscription!: Subscription;
+
+  constructor(private readonly translateService: TranslateService) {}
+
+  setSelected(selectedValue: {item: ListItem}): void {
+    const id = selectedValue?.item?.id;
+
+    if (id && typeof id === 'string') {
+      this.setSelectedValue(id);
+    }
+  }
+
+  setSelectedValue(selectedValue: SelectedValue): void {
+    this.selected$.next(selectedValue);
+  }
+
+  ngOnInit() {
+    this.setDefaultSelection();
+    this.openSelectedSubscription();
+    this.openClearSubjectSubscription();
+  }
+
+  ngOnChanges(changes: SimpleChanges): void {
+    if (
+      changes?.items?.currentValue ||
+      changes?.defaultSelectionId?.currentValue ||
+      changes?.defaultSelection?.currentValue ||
+      changes?.defaultSelectionIds?.currentValue
+    ) {
+      this.setDefaultSelection();
+    }
+  }
+
+  ngOnDestroy() {
+    this._selectedSubscription?.unsubscribe();
+    this._clearSubjectSubscription?.unsubscribe();
+  }
+
```

Comment:
Please bring in accordance to the coding guidelines

Reply:
done son